fauna/encoding/decoder.py

```diff
@@ -134,15 +137,24 @@ def _decode_dict(dct: dict, escaped: bool):
 
             if "@ref" in dct:
                 value = dct["@ref"]
+                if "id" not in value and "name" not in value:
+                    # Unsupported document reference. Return the unwrapped value to futureproof.
+                    return value
+
                 col = FaunaDecoder._decode(value["coll"])
+                doc_ref: Union[DocumentReference, NamedDocumentReference]
```

Comment:
You replaced the early return statements with this mutable variable, but still have an early return for the NullDocument case. Is this necessary?

Reply:
Only as necessary as any impl—i.e. we can write this many ways I'm sure 😄. If you want to propose a change and discuss, I'm game